<?xml version="1.0" encoding="utf-8"?>
<launch>
    <arg name="package_name" default="asv_perception_homography"/>
    <arg name="node_name" default="homography"/>
    <arg name="rgb_topic" />
    <arg name="obstacles_topic" />
    <arg name="imu_topic" />
    <arg name="points_topic" />
    <arg name="classifications_topic" />
    <arg name="points_range" />
    <arg name="calibration_vector_lb" />
    <arg name="calibration_vector_ub" />
    <arg name="use_gpu" />
    <arg name="src_frame" />
    <arg name="max_error" />

    <!-- 
        Node to compute and publish homography matrix from rgb to radar
        See homography.py
    -->
    <node name="$(arg node_name)" pkg="$(arg package_name)" type="homography.py" output="screen" launch-prefix="nice -n 10">

        <remap from="~imu" to="$(arg imu_topic)" />
        <remap from="~obstacles" to="$(arg obstacles_topic)" />
        <remap from="~points" to="$(arg points_topic)" />
        <remap from="~rgb" to="$(arg rgb_topic)" />
        <remap from="~classifications" to="$(arg classifications_topic)" />
        
        <param name="~points_range" value="$(arg points_range)" />
        <param name="~use_gpu" value="$(arg use_gpu)" />
        <param name="~calibration_lb" value="$(arg calibration_vector_lb)" />
        <param name="~calibration_ub" value="$(arg calibration_vector_ub)" />
        <param name="~src_frame" value="$(arg src_frame)" />
        <param name="~max_error" value="$(arg max_error)" />

    </node>

</launch>

